lib/livebook/evaluator.ex

```diff
@@ -391,10 +397,23 @@ defmodule Livebook.Evaluator do
     catch
       kind, error ->
         stacktrace = prune_stacktrace(__STACKTRACE__)
-        {:error, kind, error, stacktrace}
+
+        code_error =
+          if code_error?(error) and (error.file == env.file and error.file != "nofile") do
```

Comment:
@josevalim we want to ignore the error if it comes from the actual evaluation like `Code.eval_string("x")`. The file check is not a beautiful solution, but should do the job. Do you have any magic at hand? :)